### DISCUSSION TITLE
Compare fit models from envfit and ordisurf

### DISCUSSION THREAD

**Discussion by alopgar** [vegandevs/vegan#479] [Q&A]
Hi!

I have some questions about procedures to compare fitting of environmental variables through envfit and ordisurf methods. I made a PCA on microbiome datasets and have a continuous variable which I fitted this way:

```
beta.pca <- prcomp(betadat %>% dplyr::select(-one_of("NL", "DM", "CH4")))
set.seed(17)
enft <- envfit(beta.pca ~ NL + DM + CH4, data = betadat, permutations = 999)
ordsurf <- ordisurf(beta.pca, betadat.lst$CH4, plot = F, scaling = 2)
```
As I understand, `envfit` uses linear model to fit the variable into the 2 first PCs, while `ordisurf` uses a GAM to do the same. I would like to compare both models to see if my variable fits better with GLM or GAM, or if there are no differences between models. I cannot apply AIC nor Chi tests between an `envfit` and an `ordisurf` object because `envfit` is not `glm` class.
Am I approaching this well? Is there another way to compare both models?

Thanks in advance

Adrian LG

**Comment by jarioksa** [comment]
You can use `ordisurf` to fit a linear model that is identical to the linear model in `envfit` (but displayed as surface instead of an arrow of its gradient), and the function returns the `mgcv::gam` fitted object, and for this you have  an `anova` method:

```r
library(vegan)
data(varespec, varechem)
set.seed(22)
mod <- metaMDS(varespec, trace=0)
ef <- envfit(mod ~ pH, varechem) # envfit for verification
surffit <- ordisurf(mod ~ pH, varechem) # standard smooth fit
linefit <- ordisurf(mod ~ pH, varechem, knots=1) # linear surface
plot(ef, add=TRUE) # verify that envfit arrow shows the gradient (is perpendicular to contour lines)
anova(linefit, surffit, test="F") # test: linear looks quite decent
# Analysis of Deviance Table
#
# Model 1: y ~ poly(x1, 1) + poly(x2, 1)
# Model 2: y ~ s(x1, x2, k = 10, bs = "tp", fx = FALSE)
#   Resid. Df Resid. Dev     Df Deviance      F  Pr(>F)  
# 1    21.000    0.81021                                 
# 2    16.752    0.51515 4.2481  0.29507 2.5522 0.07487 .
```

**Comment by alopgar** [comment]
Very helpful, thanks!
I still have a question: with this method I can compare LM vs GAM, but only with one variable at a time. Is there any chance of building a model with covariates?
